src/remote_common.c

```diff
@@ -0,0 +1,336 @@
+/*
+ * Copyright (C) 2019 dcurl Developers.
+ * Use of this source code is governed by MIT license that can be
+ * found in the LICENSE file.
+ */
+
+#include "remote_common.h"
+#include <stdio.h>
+#include <string.h>
+
+bool die_on_amqp_error(amqp_rpc_reply_t x, char const *context)
+{
+    switch (x.reply_type) {
+    case AMQP_RESPONSE_NORMAL:
+        return true;
+
+    case AMQP_RESPONSE_NONE:
+        fprintf(stderr, "%s: missing RPC reply type!\n", context);
```

Comment:
We shall not use `stderr` because `dcurl` is intended to be a library bridged to existing applications. Instead, `ddprintf` is a wrapper for logging and diagnosis purpose.